packages/lib-sourcify/src/lib/verification.ts

```diff
@@ -59,7 +59,31 @@ export async function verifyDeployed(
     deployedBytecode,
     recompiled.immutableReferences
   );
-  if (match.status) return match;
+  if (match.status === 'perfect') return match;
+
+  if (match.status === 'partial') {
+    const checkedContractWithOriginalMetadata =
+      await checkedContract.tryToFindOriginalMetadata(deployedBytecode);
+    if (checkedContractWithOriginalMetadata) {
+      const matchWithOriginalMetadata = { ...match };
+      const recompiled = await checkedContractWithOriginalMetadata.recompile();
+
+      matchWithDeployedBytecode(
+        matchWithOriginalMetadata,
+        recompiled.deployedBytecode,
+        deployedBytecode
+      );
+      if (matchWithOriginalMetadata.status === 'perfect') {
+        checkedContract.initSolcJsonInput(
+          checkedContractWithOriginalMetadata.metadata,
+          checkedContractWithOriginalMetadata.solidity
+        );
+        return matchWithOriginalMetadata;
+      } else {
+        return match;
+      }
+    }
+  }
```

Comment:
Another thing I'm noticing now, we are assuming the verification will work with `matchWithDeployedBytecode` only. 

We should check if the match status is partial after trying all matching methods. This means if a contract can't be matched with `matchWithDeployedBytecode` but `matchWithSimulation` we miss the chance to verify it perfectly. 
    - Although we think we are not going to need "simulation" and "creatorTx" verification methods, we still don't know if they are completely useless, especially "creatorTx".

If we choose to recurse by calling `verifyDeployed` again, we should be careful not to recurse infinitely